src/main/java/jenkins/plugins/openstack/compute/JCloudsRetentionStrategy.java

```diff
@@ -58,7 +58,16 @@ private void doCheck(JCloudsComputer c) {
         final int retentionTime = node.getSlaveOptions().getRetentionTime();
         if (retentionTime <= 0) return; // 0 is handled in JCloudsComputer, negative values needs no handling
 
-        final long idleSince = c.getIdleStartMilliseconds();
+        long idleSince = c.getIdleStartMilliseconds();
+        final long connectionTime = c.getConnectionTime();
+        if(connectionTime == 0 ) {
+            //channel is already set, but connection activity is not finished completely
+            return;
+        }
+        if(connectionTime > idleSince) {
+            //idle should be counted without time consumed by establishing of connection
+            idleSince = connectionTime;
```

Comment:
I would like to see the whole computation encapusted in the `JCloudsComputer` class.